l3kernel/l3expan.dtx

```diff
@@ -125,87 +130,92 @@
+%   \cs[no-index]{\meta{name}|:|\meta{variant arg spec}} is globally
+%   created, as is the function \cs[no-index]{exp_args:N\meta{variant arg spec}}
+%   needed to expand the arguments as necessary before they
+%   are passed to the base function.
+%   So for example
+%   \begin{quote}\ttfamily
+%     |\cs_set:Npn \foo:Nn #1#2 {|\meta{code here}|}|\\
+%     |\cs_generate_variant:Nn \foo:Nn { c }|
+%   \end{quote}
+%   creates a new function |\foo:cn| which expands its first
+%   argument into a control sequence further passed as first argument to
+%   |\foo:Nn|. Similarly
+%   \begin{quote}
+%     |\cs_generate_variant:Nn \foo:Nn { NV , cV }|
+%   \end{quote}
+%   generates the variants |\foo:NV| and |\foo:cV| in the same
+%   way. Notice that
+%   \begin{quote}
+%     |\cs_generate_variant:Nn \foo:cn { cV }|
+%   \end{quote}
+%   is an error to generate |\foo:cV| because |cn| is not a
+%   \meta{base arg spec}. On the contrary
+%   \begin{quote}
+%     |\cs_generate_variant:Nn \foo:nc { Vc }|
+%   \end{quote}
+%   is allowed because |c| is not part of the \meta{base arg spec} head.
+%
+%   The \cs{cs_generate_variant:Nn} function should only be applied to
+%   already defined base functions\footnotemark{}.
+%   If the base function is protected or if the \meta{variant arg spec} contains
+%   |x|, then the variant is
+%   protected. There is no need to re-apply \cs{cs_generate_variant:Nn} after
+%   changing the definition of the base function: the variant will always
+%   use the current definition of the base.
+%   As \cs{cs_generate_variant:Nn} does nothing when the variant
+%   is already available, generating the same variants repeatedly is 
+%   safe.
 %
 %   For backward compatibility it is currently possible to make |n|,
-%   |o|, |V|, |v|, |f|, |e|, or |x|-type variants of an |N|-type argument or
-%   |N| or |c|-type variants of an |n|-type argument.  Both are
+%   |o|, |V|, |v|, |f|, |e|, or |x| type variants of an |N| type argument or
+%   |N| or |c| type variants of an |n| type argument.  Both are
 %   deprecated.  The first because passing more than one token to an
-%   |N|-type argument will typically break the parent function's code.
+%   |N| type argument will typically break the parent function's code.
 %   The second because programmers who use that most often want to
 %   access the value of a variable given its name, hence should use a
-%   |V|-type or |v|-type variant instead of |c|-type.  In those cases,
+%   |V| type or |v| type variant instead of |c| type.  In those cases,
 %   using the lower-level \cs{exp_args:No} or \cs{exp_args:Nc}
 %   functions explicitly is preferred to defining confusing variants.
 % \end{function}
+% \footnotetext{This is only
+%   enforced if debugging support \texttt{check-declarations} is enabled.}
 %
 % \begin{function}[added = 2018-04-04, updated = 2019-02-08]
 %   {\exp_args_generate:n}
 %   \begin{syntax}
 %     \cs{exp_args_generate:n} \Arg{variant argument specifiers}
```

Comment:
```suggestion
%     \cs{exp_args_generate:n}
%     ~~\{ \meta{variant arg spec_1} , \meta{variant arg spec_2} , ... \}
```

Reply:
> > % Defines function \cs[no-index]{exp_args:N\meta{var arg spec_i}} for each

BTW, `<` and `>` need to be backslash escaped, otherwise input \<...\> is eaten by GitHub's markdown parser as an HTML tag like `<head>`.